### PR TITLE
Reset to in-memory stream with empty byte sequence for None init body

### LINKED ISSUE #22304: Intermittent failure in /fetch/content-type/response.window.html
```

  ▶ Unexpected subtest result in /fetch/content-type/response.window.html:
  │ FAIL [expected NOTRUN] Request: combined response Content-Type: text/plain 
  └   → promise_test: Unhandled rejection with value: object "TypeError: Value is not valid"

  ▶ Unexpected subtest result in /fetch/content-type/response.window.html:
  │ FAIL [expected NOTRUN] Response: combined response Content-Type: text/plain 
  └   → promise_test: Unhandled rejection with value: object "TypeError: Value is not valid"

  ▶ Unexpected subtest result in /fetch/content-type/response.window.html:
  │ TIMEOUT [expected NOTRUN] Response: combined response Content-Type: text/html text/plain
  └   → Test timed out

  ▶ Unexpected subtest result in /fetch/content-type/response.window.html:
  │ FAIL [expected NOTRUN] Request: combined response Content-Type: text/html text/plain
  │   → assert_equals: expected "text/plain" but got ""
  │ 
  │ runRequestResponseTest/<@http://web-platform.test:8000/fetch/content-type/response.window.js:70:5
  │ async*Test.prototype.step@http://web-platform.test:8000/resources/testharness.js:1568:20
  │ promise_test/tests.promise_tests</<@http://web-platform.test:8000/resources/testharness.js:591:31
  └ promise_test/tests.promise_tests<@http://web-platform.test:8000/resources/testharness.js:590:20

  ▶ Unexpected subtest result in /fetch/content-type/response.window.html:
  │ FAIL [expected TIMEOUT] fetch(): combined response Content-Type: text/plain 
  │   → assert_equals: expected "text/plain" but got ""
  │ FAIL [expected NOTRUN] fetch(): separate response Content-Type: text/html text/plain
  │   → assert_equals: expected "text/plain" but got ""
  │ FAIL [expected NOTRUN] fetch(): combined response Content-Type: text/html text/plain
  │   → assert_equals: expected "text/plain" but got ""
  │ 
  │ runFetchTest/<@http://web-platform.test:8000/fetch/content-type/response.window.js:57:5
  │ async*Test.prototype.step@http://web-platform.test:8000/resources/testharness.js:1568:20
  │ promise_test/tests.promise_tests</<@http://web-platform.test:8000/resources/testharness.js:591:31
  └ promise_test/tests.promise_tests<@http://web-platform.test:8000/resources/testharness.js:590:20
```

### PR DESCRIPTION
**Pull request by CYBAI**
Because the response body stream is initialized with FetchResponse, it cannot be processed with in-memory empty sequence. Thus, instead of using the FetchResponse stream, we'll reset it to Memory body stream with empty byte sequence if there's no init body.

With this PR, we can fix the TIMEOUT in `/fetch/content-type/response.window.html` so I think it should fix #22304.

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #22304
- [x] There are tests for these changes

### CODE REVIEW

**Comment by CYBAI** [comment]
this seems to be an intermittent pass 🤔 

```
OK /css/cssom-view/elementsFromPoint-simple.html
  PASS [expected FAIL] subtest: elementsFromPoint for each corner of a simple div
```

**Comment by CYBAI** [comment]
@bors-servo retry

- there's no `fetch` or `Response` related code in the `css/cssom-view/elementsFromPoint-simple.html` test though 🤔 I may file an intermittent issue for it

**Review by mrobinson — APPROVED** [review]
This seems like a good fix to me, but I don't have a decent answer for you question. Feel free to wait to land this until you have a better answer.

**Comment by CYBAI** [comment]
> This seems like a good fix to me, but I don't have a decent answer for you question. Feel free to wait to land this until you have a better answer.

@mrobinson thanks for the review!

I'm wondering, in most cases, when `Response` constructor is used, it would be used by JS for users. When it's for `fetch` (or we say `FetchResponse`), it should only be used by the browser engine itself and users should not have the opportunity to declare a Response with `FetchResponse` stream source.

Thus, I'm wondering it's good to refactor it to use `InMemory` by default and have a new `new_for_fetch` method for `fetch`. But, if we'd like to do that, I feel like it's good to land this one and let's file a follow-up to do the refactoring so that we can fix the TIMEOUTs first and also make sure the refactoring won't break those tests again. WDYT?

Also cc @jdm for the feedback of my question and thoughts 🙏

File: components/script/dom/response.rs [hunk 1]
```diff
@@ -144,6 +144,11 @@ impl Response {
                     )?;
                 }
             };
+        } else {
+            // Reset FetchResponse to an in-memory stream with empty byte sequence here for
+            // no-init-body case
+            let stream = ReadableStream::new_from_bytes(&global, Vec::with_capacity(0));
+            r.body_stream.set(Some(&*stream));
```
[the remark this answers is not in the document]
Comment (jdm) [reply components/script/dom/response.rs:151]: Yeah, separating the code paths sounds clearer to me. I'm fine if you file a follow-up issue about it!

[the remark this answers is not in the document]
Comment (CYBAI) [reply components/script/dom/response.rs:151]: Just filed at https://github.com/servo/servo/issues/29663 🙏

**Comment by CYBAI** [comment]
@bors-servo try=wpt

- Just rebased to update test expectation for new tests in `/fetch/api/body/mime-type.any.js` but let me try again to see if there's any other test expectations I should update

**Comment by mrobinson** [comment]
@bors-servo r+

Looks good, so I'm going to go ahead and land this one.
